source/agora/consensus/state/Ledger.d

```diff
@@ -383,6 +383,17 @@ public class Ledger
 
         if (header.height == this.last_block.header.height)
             this.last_block = this.storage.readLastBlock();
+
+        // reset the cached coinbase tx if next block is coinbase block
+        // and signature is for a block in that payout period
+        const height = this.height() + 1;
+        if (height % this.params.PayoutPeriod == 0 &&
+            header.height > Height(height - 2 * this.params.PayoutPeriod)
```

Comment:
We should definitely write helpers for payout period calculations at some point, they are all over the place and pretty verbose.

and I am not sure if this is generally correct, `header.height <= Height(height - this.params.PayoutPeriod)` would not hold for the second half of the blocks that we take into account in the first payout block, no?

Reply:
I agree we need a function for this with unit test to validate it. I will add. Although I think it might be correct we will see after test is written.

Reply:
Updated and added unit test which showed I was missing the check to exclude the block height of 1 * payout period.